tests/subsys/mcuboot/ecdsa/testcase.yaml

```diff
@@ -0,0 +1,219 @@
+common:
+  tags: mcuboot
+  platform_exclude: native_posix
+  integration_platforms:
+    - nrf52840dk_nrf52840
+
+  images:
+    - name: bootloader
+      path: ../../../../../bootloader/mcuboot/boot/zephyr
+      extra_args: OVERLAY_CONFIG=../../samples/zephyr/overlay-ecdsa-p256.conf
+    - name: hello1
+      path: ../hello-world
+      extra_args: FROM_WHO=hello1
+    - name: hello2
+      path: ../hello-world
+      extra_args: FROM_WHO=hello2
+
+tests:
+  mcuboot.ecdsa.good:
+    stages:
+      - WestSign:
+          image: hello1
+          imgtool_args:
+            key: ../bootloader/mcuboot/root-ec-p256.pem
+      - WestSign:
+          image: hello2
+          imgtool_args:
+            key: ../bootloader/mcuboot/root-ec-p256.pem
+            pad: true
+      - OnTarget:
+          image: bootloader
+          harness: console
+          harness_config:
+            type: multi_line
+            regex:
+              - "Starting bootloader"
+              - "Primary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3"
+              - "Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3"
+              - "Boot source: none"
+              - "Failed reading image headers; Image=0"
+              - "Unable to find bootable image"
+      - OnTarget:
+          image: hello1
```

Comment:
I think I don't understand fully your comment. I guess getting data from dts can be done, but after discussion with mcuboot developers, we think it is better to start with user-given data and later we can enhance this to use dts.
As for combining multiple images into one: this PR is for the opposite approach. MCUboot tests require separate consecutive flashes and checking the output after each of them to monitor if the image was accepted, rejected etc.

Reply:
If you program the hex files instead of the bin files, pyocd should get the address from the hex file itself. The only thing that is needed is to erase the entire flash before the bootloader is flashed, otherwise the initial behavior is unpredictable, as it depends on what was previously in the other slots.

My comment about waiting on the DTS is mainly about trying to divide the work up. Having something able to select among targets is better than the current test, which is just hard-coded for another target. Getting the information from the DTS is even better, but something that makes more sense as a future PR.